### ISSUE TITLE
Search no longer shows submission fields under "View Options"

### ISSUE THREAD

**Issue by kmccurley** [kohler/hotcrp#242]
It seems that the logic for how search is done has changed. We used to be able to specify a field from the submission form under "View options" and have that shown in the search results, but now there is no such capability. This makes it hard for us to filter search results based on fields that were selected by the author (e.g., a checkbox for whether it is a student paper). Is this no longer available?

**Comment by kohler** [comment]
Are you running current? You often have not been.

**Comment by kmccurley** [comment]
We are running a version pulled from the last release, which was 3.01b. We are currently 66 commits behind, but the only one that seems potentially relevant is [this](https://github.com/kohler/hotcrp/commit/eb161a7bd21cf7506f3133fb62d2630bd133cf7b) but it's difficult to tell from the descriptions of the commits. What is the intent of fields that should show up in the "View options" menu? Does it include all submission form fields?

**Comment by kohler** [comment]
It does still show up there. See right column (example from HotOS)

<img width="412" alt="image" src="https://user-images.githubusercontent.com/410647/120892743-c5afa900-c5dd-11eb-8af1-0bb521bcddbb.png">

HOWEVER, in a possibly surprising UI decision, please note that if an option is missing from all the papers in a search, then View options on that search will not list the option.

**Comment by kmccurley** [comment]
I updated our version to the latest non-release version of HotCRP but it seems the problem is with our data rather than the code. We had a checkbox for major revision that was supposed to be checked by some authors, but apparently they ignored this checkbox and added comments to the program chair instead. The visibility of that field is different, so members of the program committee couldn't even see them. This is probably a problem in our configuration (or an inability of authors to follow directions). I am therefore closing this issue.

When we first set up a conference, we add extra options for special handling, we insert them into the Settings table with name='options'. I assign IDs to them at that time, but it was never clear to me if there are additional assumptions about IDs for PaperOptions - would there be any problem if there are "holes" in the option IDs? In other words, are the optionIds assumed to be sequential?
